Runtime/Plugins/Raycast/RaycastManager.cs

```diff
@@ -0,0 +1,82 @@
+using System.Collections.Generic;
+using Unity.UIWidgets.ui;
+using UnityEngine;
+using Rect = Unity.UIWidgets.ui.Rect;
+
+namespace Unity.UIWidgets.plugins.raycast {
+    public class RaycastableRect {
+        public bool isDirty;
+        public Rect rect;
+
+        public RaycastableRect(bool isDirty) {
+            this.isDirty = isDirty;
+        }
+    }
+
+    public class RaycastManager {
+        static RaycastManager _instance;
+
+        public static RaycastManager instance {
+            get {
+                if (_instance == null) {
+                    _instance = new RaycastManager();
+                }
+
+                return _instance;
+            }
+        }
+
+        public Dictionary<int, Dictionary<int, RaycastableRect>> hashCodeList =
+            new Dictionary<int, Dictionary<int, RaycastableRect>>();
+
+        public static void VerifyWindow(int windowHashCode) {
+            if (!instance.hashCodeList.ContainsKey(windowHashCode)) {
+                // Debug.Log($"New Window: @[{windowHashCode}] ({instance.hashCodeList.Count})");
+                instance.hashCodeList.Add(windowHashCode, new Dictionary<int, RaycastableRect>());
+            }
+        }
+
+        public static void AddToList(int key, int windowHashCode) {
+            VerifyWindow(windowHashCode);
```

Comment:
It may be better to register the RaycastRect List for a given window only once when the RaycastFilter is initialized, so that if a user use RaycastableContainer without RaycastFilter attached to the Panel, we should warn him/her that this is a potential issue? 
As a result, we should replace this verifyWIndow with an assertion ?

the variable "key" is not clear here, maybe we should rename it as widgetKey or something alike?

Reply:
There is a script execution order issue, in the `UIWidgetsPanel` class `OnEnable` function, the `windowAdapter` will be initialized and then `attachRootWidget`, so the `'Raycastable' widget` will be initialized first. So on the one hand when I execute `Filter` script first I can't get the `windowAdapter`, on the other hand if I execute `Filter` script after, the widget can't find an existing window key.

I don't know whether I miss or misunderstand something.

There is a way that I insert a check 'if there is an attached Filter component and pass the windowAdapter' between the `windowAdapter` initialization and the `attachRootWidget` function call.